txbridge/src/main/java/org/jboss/jbossts/txbridge/outbound/OutboundBridgeManager.java

```diff
@@ -66,18 +68,20 @@
 
         try
         {
-            Transaction transaction = (Transaction)TransactionManager.transactionManager().getTransaction();
+            Transaction transaction = (Transaction)TransactionManager.transactionManager(new InitialContext()).getTransaction();
 
-            Uid externalTxId = transaction.get_uid();
+            // generating new Uid to get it connected to incoming transaction
+            // the incoming transaction could come from different TM than Narayana is (e.g. Wildfly Transaction client)
+            Uid outboundMappingUid = new Uid();
```

Comment:
I believe this is wrong because the txId is important for ensuring we only outbound bridge once. By using a new uid each time it will not find it in the map in outboundBridgeMappings.

You can get the Uid from even a WFTC tx client by doing the following:
Uid externalTxId = new Uid(((org.wildfly.transaction.client.LocalTransaction)transaction).getXid().getGlobalTransactionId())

The reason I spotted this was because I noticed you had renamed all the variable names, I think we should revert the variable name changes.

Also, we should check what are the circumstances you might get a org.wildfly.transaction.client.RemoteTransaction because that seems to have no obvious way to get the Xid however it doesn't allow enlisting resources either so I expect it is rare to get and should not happen in this code path.